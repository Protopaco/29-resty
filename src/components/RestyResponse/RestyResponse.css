.response{
    width: 700px;
    height: 70vh;
    border: 3px solid black;
    padding: 10px;
}