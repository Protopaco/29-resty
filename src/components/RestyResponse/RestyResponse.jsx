import React, { useState, useEffect } from 'react';
import styles from './RestyResponse.css';

const RestyResponse = ({ responseObject }) => {

    return (
        <div>
            <h2>Response</h2>
            <div className={styles.response}>
                {responseObject ?
                    <div>
                        <p data-testid='responseDisplay'
                        >{JSON.stringify(responseObject)}</p>
                    </div>
                    :
                    <></>
                }
            </div>
        </div>
    )
}

export default RestyResponse;