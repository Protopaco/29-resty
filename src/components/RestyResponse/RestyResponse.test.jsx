import React from 'react';
import { render, cleanup } from '@testing-library/react';
import RestyResponse from './RestyResponse';

describe('RestyResponse component', () => {
    afterEach(() => cleanup());
    it('renders RestyResponse', () => {
        const { asFragment } = render(<RestyResponse responseObject={{ greeting: 'Hello!' }} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
