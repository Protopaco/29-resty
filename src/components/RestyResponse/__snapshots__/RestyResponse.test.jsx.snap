// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestyResponse component renders RestyResponse 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Response
    </h2>
    <div
      class="response"
    >
      <div>
        <p
          data-testid="responseDisplay"
        >
          {"greeting":"Hello!"}
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
